demo/app/android/app/src/androidTest/kotlin/integration_test.kt

```diff
@@ -50,15 +50,15 @@ class IntegrationTest {
         // Issue VCs
         val requestURI = BuildConfig.INITIATE_ISSUANCE_URL
 
-        val requiredOpenID4CIArgs = Args(requestURI, "ClientID", crypto, didResolver)
+        val requiredOpenID4CIArgs = Args(requestURI, crypto, didResolver)
 
         val ciOpts = Opts()
         ciOpts.addHeader(trace.traceHeader())
 
         val ciInteraction = Interaction(requiredOpenID4CIArgs, ciOpts)
 
-        val authorizeResult = ciInteraction.authorize()
-        assertThat(authorizeResult.userPINRequired).isFalse()
+        val pinRequired = ciInteraction.issuerCapabilities().preAuthorizedCodeGrantParams().pinRequired()
```

Comment:
Looks like we have couple of breaking API changes. Can we please add breaking change to the commit/PR title along with migration path (before and after) ?

Reply:
The examples below show just the changes for the pre-auth flow. Auth is new, so there is no "before". For more context, compare the previous examples in usage.md with the new ones.

### Before

Kotlin:
```
val args = Args("YourRequestURIHere", "ClientID", kms.getCrypto(), didResolver)

...

val result = interaction.authorize()

val credentials: CredentialsArray

if (result.UserPINRequired) {
    credentials = interaction.requestCredentialWithPIN(didDocResolution.assertionMethod(), "1234")
} else {
    credentials = interaction.requestCredential(didDocResolution.assertionMethod())
}
```

Swift:
```
let args = Openid4ciNewArgs("YourRequestURIHere", "ClientID", kms.getCrypto(), didResolver)

...

let result = interaction.authorize()

var credentials: VerifiableCredentialsArray

if result.UserPINRequired {
    credentials = interaction.requestCredential(withPIN: didDocResolution.assertionMethod(), pin:"1234")
} else {
    credentials = interaction.requestCredential(didDocResolution.assertionMethod())
}
```

### After

Kotlin:
```
val args = Args("YourRequestURIHere", kms.getCrypto(), didResolver)

...

// It's a good idea to check the issuer's capabilities first
val issuerCapabilities = interaction.issuerCapabilities()
if (!issuerCapabilities.PreAuthorizedCodeGrantTypeSupported()) {
    // This code example isn't applicable. See the authorization code flow example instead.
    
    return
}

val credentials: CredentialsArray

if (issuerCapabilities.preAuthorizedCodeGrantParams().pinRequired()) {
    credentials = interaction.requestCredentialWithPIN(didDocResolution.assertionMethod(), "1234")
} else {
    credentials = interaction.requestCredential(didDocResolution.assertionMethod())
}
```

Swift:
```
let args = Openid4ciNewArgs("YourRequestURIHere", kms.getCrypto(), didResolver)

...

// It's a good idea to check the issuer's capabilities first
let issuerCapabilities = interaction.issuerCapabilities()
if !issuerCapabilities.PreAuthorizedCodeGrantTypeSupported() {
    // This code example isn't applicable. See the authorization code flow example instead.
    
    return
}

var credentials: VerifiableCredentialsArray

if issuerCapabilities.preAuthorizedCodeGrantParams().pinRequired() {
    credentials = interaction.requestCredential(withPIN: didDocResolution.assertionMethod(), pin:"1234")
} else {
    credentials = interaction.requestCredential(didDocResolution.assertionMethod())
}
```

Reply:
Thanks - copied to PR description.